os/kernel/log_dump/log_dump.c

```diff
@@ -125,8 +130,28 @@ static void log_dump_mem_check(size_t max_size)
 	}
 }
 
+size_t log_dump_set_option(FAR const char *buffer, size_t buflen)
+{
+	(void)buflen;
+
+	if (strncmp(buffer, LOGDUMP_SAVE_START, strlen(LOGDUMP_SAVE_START) + 1) == 0) {
+		is_started_to_save = true;
+	} else if (strncmp(buffer, LOGDUMP_SAVE_STOP, strlen(LOGDUMP_SAVE_STOP) + 1) == 0) {
+		is_started_to_save = false;
+	} else {
+		return LOG_DUMP_OPT_FAIL;
+	}
+	return LOG_DUMP_OK;
+}
+
+
+
 int log_dump_save(char ch)
 {
+	if (is_started_to_save == false) {
+		return LOG_DUMP_OK;
```

Comment:
In my opinion, saved log should be cleaned here, because user can stop logging before read it.

Reply:
From the requirement, user wants to stop to save the log before read, and log will be saved by default after boot up.
So if we clean the saved log in here, we can miss some logs.